llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -8170,11 +8170,17 @@ SDValue RISCVTargetLowering::LowerOperation(SDValue Op,
     return lowerRESET_FPENV(Op, DAG);
   case ISD::EH_DWARF_CFA:
     return lowerEH_DWARF_CFA(Op, DAG);
+  case ISD::VP_SELECT:
+    if (SDValue Op2 = Op.getOperand(2);
```

Comment:
Have you tried doing this as a pre-lowering combine instead?  I could see that triggering other interactions, but it would address the missing fold problem Craig notes below.

Reply:
Indeed,  pre-lowering combine is a better,  it reuses the optimization from the SDNode, and I've already made the change.

Reply:
I think by "pre-lowering combine", @preames meant called for `PerformDAGCombine` not `LowerOperation`

Reply:
I previously implemented a version using combinehttps://github.com/llvm/llvm-project/commit/3151db1ab319a014579d5b96b77c7f5e44789319, but then thought lowering would be more appropriate and switched to that，now I'm about to switch it back. thanks